### ISSUE TITLE
Mono 6 is out and OpenSim runs great on it

### ISSUE THREAD

**Issue by Misterblue** [Misterblue/opensim-docker#3]
The Dockerfiles use Mono 5 which is outdated and Mono6 has proved good for running OpenSimulator on Linux.

**Comment by echads** [comment]
Yes I was just playing around and noticed two errors in Mono 5.
#
At the `msbuild` stage on 5.20.1.34 and 5.18.1.0 regarding a missing assembly System.Security.Cryptography.X509Certificates and MemAssetService not fully implementing the IAssetService interface. It looks like these things are on your radar.
```
Build FAILED.

"/home/opensim/opensim/OpenSim.sln" (default target) (1) ->
"/home/opensim/opensim/addon-modules/HerbalTransportCS/HerbalTransportCS/Herbal3d.HerbalTransportCS.csproj" (default target) (25) ->
(ResolveAssemblyReferences target) -> 
  /usr/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(2130,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Security.Cryptography.X509Certificates". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/home/opensim/opensim/addon-modules/HerbalTransportCS/HerbalTransportCS/Herbal3d.HerbalTransportCS.csproj]


"/home/opensim/opensim/OpenSim.sln" (default target) (1) ->
"/home/opensim/opensim/addon-modules/HerbalCommonEntitiesCS/CommonEntitiesConv/Herbal3d.CommonEntitiesConv.csproj" (default target) (4) ->
(CoreCompile target) -> 
  MemAssetService.cs(29,36): error CS0535: 'MemAssetService' does not implement interface member 'IAssetService.Get(string, string, bool, SimpleAssetRetrieved)' [/home/opensim/opensim/addon-modules/HerbalCommonEntitiesCS/CommonEntitiesConv/Herbal3d.CommonEntitiesConv.csproj]


"/home/opensim/opensim/OpenSim.sln" (default target) (1) ->
"/home/opensim/opensim/addon-modules/HerbalTransportCS/HerbalTransportCS/Herbal3d.HerbalTransportCS.csproj" (default target) (25) ->
  ServerListener.cs(119,43): error CS0246: The type or namespace name 'X509Certificate2' could not be found (are you missing a using directive or an assembly reference?) [/home/opensim/opensim/addon-modules/HerbalTransportCS/HerbalTransportCS/Herbal3d.HerbalTransportCS.csproj]

    1 Warning(s)
    2 Error(s)

```

**Comment by Misterblue** [comment]
closed by 2.1.3
